src/reducers/starWarsReducer.js

```diff
@@ -1,13 +1,28 @@
-import /* we need our action types here*/ "../actions";
+import {
+  /* we need our action types here*/
+  FETCH_SWCHARS_FETCHING,
+  FETCH_SWCHARS_SUCCESS,
+  FETCH_SWCHARS_FAILURE
+} from '../actions';
+
 const initialState = {
-  characters: []
+  characters: [],
   // Array characters, Boolean fetching, null error.
+  fetching: false,
+  error: null
 };
+
 export const charsReducer = (state = initialState, action) => {
   switch (action.type) {
     // Fill me in with the important reducers
     // action types should be FETCHING, SUCCESS and FAILURE
     // your switch statement should handle all of these cases.
+    case FETCH_SWCHARS_FETCHING:
+      return { ...state, fetching: true, error: null };
+    case FETCH_SWCHARS_SUCCESS:
+      return { ...state, characters: action.payload, fetching: false };
```

Comment:
yes, this functions (this is going to be another "if your app expands" correction, nothing wrong with the functionality as it is now). however, if your app were to expand -- let's say you added the functionality of being able to add a new character to this list -- you would want to spread the payload in after the original array, even though here it's empty. so: `{...state, character: [...state.characters, ...action.payload], fetching: false}`